src/goto-programs/lazy_goto_model.cpp

```diff
@@ -122,21 +126,17 @@ void lazy_goto_modelt::initialize(const cmdlinet &cmdline)
 
       if(!infile)
       {
-        msg.error() << "failed to open input file `" << filename
-                    << '\'' << messaget::eom;
-        throw 0;
+        throw system_exceptiont(
+          "failed to open input file `" + filename + '\'');
```

Comment:
It's quite possible that the root cause is because the user passed in the wrong filename. Or it could fail for other reasons. The only thing we know at this point is that we can't open the file, so we report that as such here.